employee-rostering-frontend/src/store/tenant/reducers.ts

```diff
@@ -17,8 +17,14 @@
 import { ActionType, TenantData, TenantAction } from './types';
 import { withElement, withoutElement } from 'util/ImmutableCollectionOperations';
 
+const path = window.location.pathname;
+let windowTenantId: number | null = null;
+if (path.indexOf('/', 1) > 0) {
+  windowTenantId = parseInt(path.substring(1, path.indexOf('/', 1)));
```

Comment:
Not nice for 2 reasons:
1. Breaks one of the [core principles](https://redux.js.org/introduction/three-principles#changes-are-made-with-pure-functions) of Redux by making the reducer impure. [Reducers](https://redux.js.org/glossary#reducer) must be pure functions—functions that return the exact same output for given inputs.
2. Mixing browser-specific code into app state management code that's meant to be UI-agnostic (https://www.quora.com/Can-Redux-be-used-for-server-side-Node-js-without-any-UI).

I suggest moving this to `index.tsx` where you call `configureStore()` and pass this information as the `preloadedState` argument.

Reply:
1. Reducer is still pure: this affects only the initial state, and the reducer only references current state + params.

2. That I am okay with

The problem with moving to `index.tsx` is there is no way to actually set initial state in `configureStore`, although using actions might work but be inefficent/make extra calls

Reply:
1. You're right. I didn't realize that `windowTenantId` is assigned only once when the file is loaded and it will (probably) never change (would be better to make it `const` to make it obvious that it never changes). Note that `tenantReducer()` function does reference the initial state in the case when the state argument us `undefined`.

2. I thought the state can be (partially) hydrated using the `preloadedState` argument of [createStore()](https://redux.js.org/api/createstore).

Reply:
In regards to 2., preloadedState require an entire state, no partials allowed unfortunately.

Reply:
2. Can be done (tried locally). Consider optional for this PR. I can submit the change later.

Reply:
Actually, partials are allowed: https://github.com/reduxjs/redux/blob/037ea567e9808ef4ad281f3e19bb00c1e74587ba/index.d.ts#L211.

Reply:
Moved it to `index.tsx` (although beware: the partial you passed in is not merged with the initial state)